specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1170 @@
+    .GetXml();
+```
+
+Result from GetXml():
+
+```c#
+<toast launch="action=openThread&amp;threadId=92187" displayTimestamp="2022-06-19T08:30:00Z">
+    <visual>
+        <binding template="ToastGeneric">
+            <text hint-maxLines="1">Jill Bender</text>
+            <text>Check out where we camped last weekend! It was incredible, wish you could have come on the backpacking trip!</text>
+            <image placement="appLogoOverride" hint-crop="circle" src="https://unsplash.it/64?image=1027"/>
+            <image placement="hero" src="https://unsplash.it/360/180?image=1043"/>
+        </binding>
+    </visual>
+    <actions>
+        <input id="textBox" type="text" placeHolderContent="reply"/>
+        <action
+          content="Send"
+          imageUri="Assets/Icons/send.png"
+          hint-inputId="textBox"
+          arguments="action=reply&amp;threadId=92187"/>
+    </actions>
+</toast>
+```
+
+## Limitations
+
+-   AppNotificationScenario::Urgent is only supported for to builds 19041 (20H1) and above. Using
+    this attribute on lower builds will not prevent the AppNotification from being constructed, but
+    developers will receive a debug warning that the attribute is not supported.
+
+# Text
+
+The Text component sets up the xml for a \<text\> element. Developers can define the content,
+language, and placement of the text on the AppNotification.
+
+**WinAppSDK 1.2 \<text\> Schema**:
+
+```c#
+<text lang? = string
+      placement? = "attribution"
+      hint-callScenarioCenterAlign? =
+      hint-maxLines? = integer />
+```
+
+These attributes are abstracted with the AppNotificationTextProperties component APIs:
+
+-   SetLanguage(String language)
+-   SetHintMaxLines(Int32)
+-   UseCallScenarioAlign()
+
+Below are some example usages:
+
+![Text Example](TextExample.png)
+
+AppNotificationContent Code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .SetAttributionText(L"Attribution Text")
+    .GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+      <text placement="attribution">Attribution Text</text>
+    </binding>
+  </visual>
+</toast>
+```
+
+Developers can use UseCallScenarioAlign() to center the text like an incoming call. This feature
+will be ignored unless the AppNotification scenario is set to IncomingCall using
+AppNotificationContent::SetScenario(enum AppNotificationScenario).
+
+![IncomingCall Simple Example](IncomingCallSimpleExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .SetScenario(AppNotificationScenario::IncomingCall)
+    .AddText(L"Incoming Call", AppNotificationTextProperties()
+        .UseCallScenarioAlign())
+    .AddText(L"Andrew Barnes", AppNotificationTextProperties()
+        .UseCallScenarioAlign())
+    .AddInlineImage(winrt::Windows::Foundation::Uri(L"https://unsplash.it/100?image=883"), ImageCrop::Circle)
+    .AddButton(Button()
```

Comment:
Did we settle on calling this Button()? Should this be AppNotificationButton() to avoid confusion with the UI Button?

Reply:
I don't see the confusion. The type it takes is AppNotificationButton - that should be enough. Don't make the names longer than needed. This short name makes the most sense, and what you'd expect it is called when adding a button

Reply:
Then this snippet needs to be updated to .AddButton(AppNotificationButton() .. right?